repository.go

```diff
@@ -46,47 +47,54 @@ func (r *Repository) CloneInto(path string, bare bool) error {
 	})
 
 	if err == git.ErrRepositoryAlreadyExists {
-		// Pull instead of clone
-		if gitRepo, err = git.PlainOpen(path); err == nil {
-			// we need to check whether it's a bare repo or not.
-			// if not we should pull, if it is then pull won't work
-			if isBare, bErr := isBare(gitRepo); bErr == nil && !isBare {
-				if w, wErr := gitRepo.Worktree(); wErr != nil {
-					err = wErr
-				} else {
-					err = w.Pull(&git.PullOptions{
-						Auth:     auth,
-						Progress: os.Stdout,
-					})
-				}
-			}
+		// as the repo already exists, we just plain open it
+		gitRepo, err = git.PlainOpen(path)
+		if err != nil {
+			return err
 		}
 	}
 
-	switch err {
-	case transport.ErrEmptyRemoteRepository:
-		log.Printf("%s is an empty repository", r.FullName)
-		//  Empty repo does not need backup
-		return nil
-	default:
+	var remotes []*git.Remote // we only create this variable here so we don't have to use := with the Remotes() call, which would create a new err
+	remotes, err = gitRepo.Remotes()
+	if err != nil {
 		return err
-	case git.NoErrAlreadyUpToDate:
-		log.Printf("No need to pull, %s is already up-to-date", r.FullName)
-		// Already up to date on current branch, still need to refresh other branches
-		fallthrough
-	case nil:
-		// No errors, continue
-		err = gitRepo.Fetch(&git.FetchOptions{
+	}
+
+	// we iterate over all the remotes, and we fetch all their heads
+	for _, remote := range remotes {
+		err = remote.Fetch(&git.FetchOptions{
 			Auth:     auth,
 			Progress: os.Stdout,
 			Tags:     git.AllTags,
 			Force:    true,
+			RefSpecs: []config.RefSpec{"+refs/heads/*:refs/heads/*"},
+		})
+		if err != nil && err != git.NoErrAlreadyUpToDate {
+			break
+		}
+	}
+
+	if !isBare(gitRepo) {
+		var worktree *git.Worktree // don't wanna shadow err
+		worktree, err = gitRepo.Worktree()
+		if err != nil {
+			return err
+		}
+
+		// this should probably be replaced with a Checkout call instead
+		err = worktree.Pull(&git.PullOptions{
```

Comment:
So ideally I would use https://pkg.go.dev/github.com/go-git/go-git/v5#Worktree.Checkout here, I just couldn't figure out a nice way to do this right away.

Reply:
I don't think we have to wait for the change, the overhead of a non-change pull is negligible unless you're updating every minute but that's not exactly the main goal of this tool. But the decision is yours. Once the PR out of draft I will take a closer look.